### ISSUE TITLE
Config file does not accept environment variables in Windows

### ISSUE THREAD

**Issue by 5chdn** [paritytech/parity#6417]
_Before filing a new issue, please **provide the following information**._

> I'm running:
>
> - **Parity version**: 1.7.0
> - **Operating system**: Windows
> - **And installed**: via installer

_Your issue description goes here below. Try to include **actual** vs. **expected behavior** and **steps to reproduce** the issue._

Split from #6405. Panic:

    C:\Program Files\Parity\Parity>parity.exe ui
    Loading config file from C:\Users\micah\AppData\Roaming\Parity\Ethereum\config.toml
    thread 'main' panicked at 'Logger is initialized only once; qed: "Cannot write to log file given: %TEMP%\\Parity\\debug.log"', src\libcore\result.rs:859
    stack backtrace:
       0: <unknown>
       1: <unknown>
       2: <unknown>
       3: <unknown>
       4: <unknown>
       5: <unknown>
       6: <unknown>
       7: <unknown>
       8: <unknown>
       9: <unknown>
      10: <unknown>
      11: <unknown>
      12: <unknown>
      13: <unknown>
      14: hid_error
      15: BaseThreadInitThunk

Config:

    [parity]
    chain = "foundation"

    [misc]
    logging = "rpc=trace"
    log_file = "%TEMP%\\Parity\\debug.log"

**Comment by 5chdn** [comment]
Same applies for macos.

**Comment by tomusdrw** [comment]
Environment variables in CLI are resolved by the terminal, for configs we would need to build our own parser to support that.
While I agree that it shouldn't panic if the logger cannot be initialized supporting env variables is a bit more involved **enhancement** or perhaps **annoyance**.

**Comment by debris** [comment]
in #8342 we added support for `$HOME` and `$BASE` variables. I believe we'll not do anything else so I'm closing this issue
